src/components/ForecastMainBanner.svelte

```diff
@@ -0,0 +1,112 @@
+<script>
+  import { Spinner } from 'flowbite-svelte';
+
+  import thermostat from '../assets/thermostat.svg';
+  import arrow from '../assets/arrow.svg';
+  import umbrella from '../assets/umbrella.svg';
+  import air from '../assets/air.svg';
+  import clock from '../assets/clock.svg';
+  import weatherCodes from '../assets/weather-interpretation-code-description.json';
+  import getDayPeriod from '../utils/dayPeriod';
+
+  export let weatherData = {};
```

Comment:
Maybe better use approach like `export let { time, isLoading, hasError, weatherCode, temperature, precipitation, feelsLike ...}` and `<ForecastMainBanner {...weatherData} />
` it's more implicate what props it receive and avoids duplication of `weatherData`